libkbfs/md_journal.go

```diff
@@ -464,7 +464,10 @@ func (j *mdJournal) convertToBranch(
 		brmd.SetUnmerged()
 		brmd.SetBranchID(bid)
 
-		// Delete the old "merged" version from the cache.
+		// Delete the old "merged" version from the cache.  We aren't
+		// equipped here to cache the new version, but it will
+		// eventually get passed via an mdFlushListener to somewhere
+		// that will cache it.
```

Comment:
That was one of my original plans, but I got stuck at the "if it doesn't exist, continue" part.  Seems like we'd still need the code I have written to make it work in every case, right?

Agreed it's not ideal to put it in the cache out-of-band, but I can't think of a better way that wouldn't require a full copy of the Config (or many new components) to be plumbed here, which you seem to be against.